include/aws/mqtt/private/client_impl.h

```diff
@@ -236,6 +272,12 @@ struct aws_mqtt_client_connection {
          * Helps us find the next free ID faster.
          */
         uint16_t packet_id;
+
+        /**
+         * Statistics tracking operational state
+         */
+        struct aws_mqtt_connection_operation_statistics_impl operation_statistics_impl;
```

Comment:
A little confused why we would put a bunch of atomics in the shared state.  We can use atomics and not bother with any locks or shared state at the cost of perfect consistency (which is what the 5 client does).  Or we can use size_ts in the shared state with locking.  But not a mixture.